src/Network/Network/NatGateway/GetAzureRMNatGatewayCommand.cs

```diff
@@ -0,0 +1,137 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using AutoMapper;
+using CNM = Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NatGateway", DefaultParameterSetName = "NoExpand"), OutputType(typeof(PSNatGateway))]
+    public partial class GetAzureRmNatGateway : NetworkBaseCmdlet
+    {
+        private const string ListParameterSet = "ListParameterSet";
+        private const string GetByNameParameterSet = "GetByNameParameterSet";
+        private const string GetByResourceIdParameterSet = "GetByResourceIdParameterSet";
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The resource group name of the nat gateway.",
+            ParameterSetName = GetByNameParameterSet,
+            ValueFromPipelineByPropertyName = true)]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The resource group name of the nat gateway.",
+            ParameterSetName = ListParameterSet,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Alias("ResourceName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the nat gateway.",
+            ParameterSetName = GetByNameParameterSet,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceNameCompleter("Microsoft.Network/natGateways", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Nat Gateway Id",
+            ParameterSetName = GetByResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string NatGatewayId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The resource reference to be expanded.",
+            ParameterSetName = "Expand",
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string ExpandResource { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (this.IsParameterBound(c => c.NatGatewayId))
+            {
+                var resourceIdentifier = new ResourceIdentifier(this.NatGatewayId);
+                this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                this.Name = resourceIdentifier.ResourceName;
+            }
+
+            if (!string.IsNullOrEmpty(this.Name))
```

Comment:
Please change to `if (ShouldGetByName(this.ResourceGroupName, this.Name))` to support wildcards properly